internal/checker/checker.go

```diff
@@ -64,44 +68,88 @@ func (c *checker) Start() {
 		if err != nil {
 			log.Printf("Error fetching message: %v\n", err)
 		}
-		for _, url := range message.Value.URLs {
-			err := c.ErrorBasedCheck(url)
-			if err != nil {
-				log.Printf("Error-based check error:%v\n", err)
-			}
+		result := c.processMessage(message)
+		result.Result = deleteEmpty(result.Result)
+		if len(result.Result) == 0 {
+			result.Result = append(result.Result, "SQLi-check: No vulnerabilities found.")
+		}
+		err = c.sendResult(result)
+		if err != nil {
+			log.Println(err)
+		} else {
+			log.Printf("Results have been successfully sent to Result Collector.")
+		}
+
+	}
+}
+
+// processMessage starts checking given list of url, appending results received.
+func (c *checker) processMessage(m broker.Message) *result.Result {
+	results := make([]string, 0)
+	totalResult := result.Result{TaskID: m.Value.ID, Result: make([]string, 0)}
+
+	for _, url := range m.Value.URLs {
+		res, err := c.ErrorBasedCheck(url)
+		if err != nil {
+			log.Printf("Error-based check:%v\n", err)
 		}
-		// set it to nil for stop processing previous result while no messages from Kafka.
+		results = append(results, res...)
+	}
+	totalResult.Result = append(totalResult.Result, results...)
+	return &totalResult
+}
+
+// sendResult sends given results via gRPC.
+func (c *checker) sendResult(r *result.Result) error {
+	conn, err := grpc.Dial(c.config.GRPCConfig.ResultCollectorAddress, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
```

Comment:
Do we really need to open a connection every time we send? You can add a client to the checker struct